src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapPlot.jsx

```diff
@@ -121,83 +122,148 @@ const HeatmapPlot = (props) => {
     }
   }, [width]);
 
-  const downsample = (groupBy) => {
+  // Defines sorting order according to which group the cell belongs to
+  const positionInGrouping = (cellSetNames, cell) => {
+    let position = 0;
+
+    for (const { key: cellSetName } of cellSetNames) {
+      if (cellsInSet(cellSetName).has(cell)) {
+        return position;
+      }
+
+      position += 1;
+    }
+
+    // If not in grouping set it in the end
+    return position + 1;
+  };
+
+  const cellsInSet = (cellSetName) => {
+    return properties[cellSetName].cellIds;
+  };
+
+  ///////////////////////////////////////////////////////////////////////////////////////////
+  // Returns a set of all cells relevant to the experiment (so the ones belonging to one or more of the enabled sets and not hidden).
+  const getAllRelevantCellIds = (arrayOfcellSetNamesByGroupByKey) => {
```

Comment:
intead of this name and the comment above, it will be more clean and easy to read if you just rename function `getAllRelevantCellIds` to `getNonHiddenCellIds`.

Reply:
yes, I was trying to rename everything from rootNode to this but it is probably too wordy.

By the way, it doesn't only filter out hidden cellIds, it also avoids picking cellIds that are not in any cellSet defined by the enabled groupBys, so nonHidden wouldn't really represent this function either.

I could just separate the functionality and have one function that filters out hidden cellIds and another with this name: `getCellIdsInGroupBys`.

Let me know what you think about this.

Reply:
and about the other comment you're right, it is better to just use the `getCellSetNamesDict` inside the function that needs it.

Reply:
Thanks for clarification Martin, I see why you used the `relevant` word in the name now. Maybe `getEnabledCellIds` would be better name then? Or anything else if you think about that is not generic and quite short